### PR TITLE
feat(header): new flag to add a header to each generated file

### PR DESCRIPTION
**Pull request by ohler**

### CODE REVIEW

File: lib/core/types.ts [hunk 1]
```diff
@@ -8,6 +8,7 @@ export interface MainOptions extends Options {
   exportTypeName: string;
   exportTypeInterface: string;
   listDifferent: boolean;
+  header?: string;
```
Comment (skovy) [inline lib/core/types.ts:11]: Even though it's optional via the CLI, I believe this value always exists so we shouldn't need to mark it as optional.

```suggestion
  header: string;
```

Comment (ohler) [reply lib/core/types.ts:11]: it's true that the value is always set when called from the CLI code, but I think making it required is a worse internal API.  the header really is optional, allowing callers to omit the argument creates no ambiguity.  making it required complicates the code; not in a huge way, but various tests would need updating with an extra line of boilerplate.

up to you though, I'll add the extra code if you don't find this convincing.

Comment (skovy) [reply lib/core/types.ts:11]: Yea, I agree updating tests is unfortunate and not ideal. However, the other arguments are already using this pattern so I think keeping this consistent is better than mixing both approaches. If you'd like to explore making all of these arguments optional in a follow-up PR I would be very happy to review that.

**Comment by skovy** [comment]
Hey @ohler, just checking if you wanted me to make a few updates and get this merged, or close out the PR?

**Comment by skovy** [comment]
Closing because this PR is stale and another PR has been opened to add this functionality.

**Pull request closed without merging**